### PR TITLE
Fix element services and AMP.setState in FIE

### PR DESCRIPTION
**Pull request by choumx**
- Minimal fix to get element services (e.g. amp-bind, amp-gwd-animation) working in FIE (broken in #10992 I think).
- Fixes `AMP.setState` in FIE (broken in #16328).

QA=Tested amp-bind page in local A4A harness.

Follow-ups:
- [ ] Ideally we can install an element service in FIE without installing it on the parent. Proposal: add `static function installServiceInEmbedWin()` to `EmbeddableService` interface.
- [ ] Add integration tests for amp-bind/amp-gwd-animation in FIE to prevent future regressions.
- [ ] #16322

### CODE REVIEW

**Comment by dvoytenko** [comment]
@choumx Doesn't `adoptEmbedWindow` and `installServiceInEmbedScope` not solve this already?

File: src/service/extensions-impl.js [hunk 1]
```diff
@@ -454,6 +457,20 @@ export class Extensions {
...
+        // installing it in the AmpDoc. See #19344.
+        const frameElement = /** @type {!Node} */ (dev().assert(
+            childWin.frameElement, 'frameElement not found for embed'));
+        const ampdoc = getAmpdoc(frameElement);
+        this.installExtensionInDoc_(ampdoc, extensionId);
+        return extension;
+      }).then(extension => {
         // Adopt embeddable extension services.
         extension.services.forEach(service => {
           adoptServiceForEmbedIfEmbeddable(childWin, service);
```
Comment (choumx) [inline src/service/extensions-impl.js]: @dvoytenko `adoptEmbedWindow` and `installServiceInEmbedScope` are called from here. The problem is this:
https://github.com/ampproject/amphtml/blob/7731c43f2c7d01da647e31ccf61b85e773d8e4b7/src/service.js#L655-L663

`adoptServiceForEmbedIfEmbeddable` early-outs if the service is not registered on the ampdoc. 

The short fix (this PR) is to register the service on the ampdoc before this happens. This means that the parent will always have _all_ services from its children FIEs.

The long fix is to decouple this. We should make it easier to install services on the FIE (see the follow-up in PR description). 

I'd like to merge this short-fix first to unblock amp-gwd-animation and amp-bind in FIE.

Comment (dvoytenko) [reply src/service/extensions-impl.js]: @choumx Apologies - got caught up with some urgent holiday releases. Your explanation makes sense. It does point to the further urgency to refactor FIE to assign `ampdoc` on FIE level. Can we discuss what remains of that work?

Comment (choumx) [reply src/service/extensions-impl.js]: > It does point to the further urgency to refactor FIE to assign ampdoc on FIE level.

Completely agree. My thoughts are we can quickly mitigate by fixing the user-facing issues (APIs) in the short-term. Then, more slowly refactor/simplify under the hood in the long-term.

Specifically: in the rest of Q4, I'd like to finish the follow-up tasks in this PR and refactor the service APIs as described in #16322. That should address the largest user-facing issues for "FIE don't have an AmpDoc".

**Pull request merged**